.github/workflows/build-and-push-images.yaml

```diff
@@ -59,7 +60,63 @@ jobs:
         if: steps.publish.outcome == 'skipped'
         uses: ./.github/workflows/template-publish-image
         with:
-          image: docker.io/kubeflow/${{ matrix.component-name }}
+          image: ghcr.io/kubeflow/trainer/${{ matrix.component-name }}
+          dockerfile: ${{ matrix.dockerfile }}
+          platforms: ${{ matrix.platforms }}
+          context: ${{ matrix.context }}
+          push: false
+
+  build-and-publish-to-dockerhub:
```

Comment:
I think this and GHCR can consolidate into one.

```
- name: Checkout
- name: GHCR Login
- name: Docker Hub Login
- name: Publish Component ${{ matrix.component-name }}
- name: Test Build For Component ${{ matrix.component-name }}
```

Because `images` can be specified multiple name: https://github.com/docker/metadata-action?tab=readme-ov-file#inputs

@mahdikhashan could you open an issue or work on this improvement?

Reply:
you are right, i'm more leaned toward current change since it separates path for publishing and has less mental effort for contribution. final decision is yours, i can merge them in this pr.

Reply:
> you are right, i'm more leaned toward current change since it separates path for publishing and has less mental effort for contribution. final decision is yours, i can merge them in this pr.

I do not want to block this PR by this my comment. So, I'm happy to work on my additional suggestion in the next PR, thanks.

Reply:
/hold cancel

Reply:
Please let me just know if you can work on it or just open an issue.